<a href="https://colab.research.google.com/github/carlosrodgu/AmigoSecreto/blob/main/Desafio_Telexom_X.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Descripción**

## **Telecom X - Análisis de Evasión de Clientes**

Has sido contratado como asistente de análisis de datos en Telecom X y formarás parte del proyecto "Churn de Clientes". La empresa enfrenta una alta tasa de cancelaciones y necesita comprender los factores que llevan a la pérdida de clientes.

Tu desafío será recopilar, procesar y analizar los datos, utilizando Python y sus principales bibliotecas para extraer información valiosa. A partir de tu análisis, el equipo de Data Science podrá avanzar en modelos predictivos y desarrollar estrategias para reducir la evasión.

¿Qué vas a practicar?

✅ Importar y manipular datos desde una API de manera eficiente.

✅ Aplicar los conceptos de ETL (Extracción, Transformación y Carga) en la preparación de los datos.

✅ Crear visualizaciones estratégicas para identificar patrones y tendencias.

✅ Realizar un Análisis Exploratorio de Datos (EDA) y generar un informe con insights relevantes.

¡Ahora es tu turno! 🚀 Usa tus conocimientos para transformar datos en información estratégica y ayudar a Telecom X a retener más clientes.



# **Descripción**

Para iniciar tu análisis, necesitarás importar los datos de la API de Telecom X. Estos datos están disponibles en formato JSON y contienen información esencial sobre los clientes, incluyendo datos demográficos, tipo de servicio contratado y estado de evasión.

📌 Enlace de la API:
🔗 challenge2-data-science-LATAM/TelecomX_Data.json at main · ingridcristh/challenge2-data-science-LATAM

🔗GitHub - ingridcristh/challenge2-data-science-LATAM

¿Qué debes hacer?

✅ Cargar los datos directamente desde la API utilizando Python.

✅ Convertir los datos a un DataFrame de Pandas para facilitar su manipulación.

Este es el primer paso para transformar los datos en información valiosa. ¿Listo para programar? 🚀

In [1]:
#importando librerías
import pandas as pd

#Cargando la data a un dataframe
url='https://raw.githubusercontent.com/ingridcristh/challenge2-data-science-LATAM/refs/heads/main/TelecomX_Data.json'
df=pd.read_json(url)
type(df)

pandas.core.frame.DataFrame

In [2]:
df.dtypes

,0
customerID,object
Churn,object
customer,object
phone,object
internet,object
account,object


In [3]:
#Analizar los tipos de datos de las columnas
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   customerID  7267 non-null   object
 1   Churn       7267 non-null   object
 2   customer    7267 non-null   object
 3   phone       7267 non-null   object
 4   internet    7267 non-null   object
 5   account     7267 non-null   object
dtypes: object(6)
memory usage: 340.8+ KB


In [4]:
df.shape


(7267, 6)

In [5]:
df

,customerID,Churn,customer,phone,internet,account
0,0002-ORFBO,No,"{'gender': 'Female', 'SeniorCitizen': 0, 'Part...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'One year', 'PaperlessBilling': '..."
1,0003-MKNFE,No,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'Yes'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
2,0004-TLHLJ,Yes,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
3,0011-IGKFF,Yes,"{'gender': 'Male', 'SeniorCitizen': 1, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
4,0013-EXCHZ,Yes,"{'gender': 'Female', 'SeniorCitizen': 1, 'Part...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
...,...,...,...,...,...,...
7262,9987-LUTYD,No,"{'gender': 'Female', 'SeniorCitizen': 0, 'Part...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'One year', 'PaperlessBilling': '..."
7263,9992-RRAMN,Yes,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'Yes'}","{'InternetService': 'Fiber optic', 'OnlineSecu...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
7264,9992-UJOEL,No,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'Month-to-month', 'PaperlessBilli..."
7265,9993-LHIEB,No,"{'gender': 'Male', 'SeniorCitizen': 0, 'Partne...","{'PhoneService': 'Yes', 'MultipleLines': 'No'}","{'InternetService': 'DSL', 'OnlineSecurity': '...","{'Contract': 'Two year', 'PaperlessBilling': '..."


In [6]:
#Analizar si existen datos nulos
df.isnull().sum()

,0
customerID,0
Churn,0
customer,0
phone,0
internet,0
account,0


In [7]:
#Normalizaremos las columnas que traen diccionarios
#Crearemos DataFrames independientes para cada categoría
df_customer = pd.json_normalize(df['customer'])
df_phone = pd.json_normalize(df['phone'])
df_internet = pd.json_normalize(df['internet'])
df_account = pd.json_normalize(df['account'])

In [8]:
# Concatenamos todo en un solo DataFrame limpio
# Mantenemos 'customerID' y 'Churn' del original
df_final = pd.concat([df[['customerID', 'Churn']], df_customer, df_phone, df_internet, df_account], axis=1)

In [9]:
df_final

,customerID,Churn,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,...,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,Charges.Monthly,Charges.Total
0,0002-ORFBO,No,Female,0,Yes,Yes,9,Yes,No,DSL,...,Yes,No,Yes,Yes,No,One year,Yes,Mailed check,65.60,593.3
1,0003-MKNFE,No,Male,0,No,No,9,Yes,Yes,DSL,...,No,No,No,No,Yes,Month-to-month,No,Mailed check,59.90,542.4
2,0004-TLHLJ,Yes,Male,0,No,No,4,Yes,No,Fiber optic,...,No,Yes,No,No,No,Month-to-month,Yes,Electronic check,73.90,280.85
3,0011-IGKFF,Yes,Male,1,Yes,No,13,Yes,No,Fiber optic,...,Yes,Yes,No,Yes,Yes,Month-to-month,Yes,Electronic check,98.00,1237.85
4,0013-EXCHZ,Yes,Female,1,Yes,No,3,Yes,No,Fiber optic,...,No,No,Yes,Yes,No,Month-to-month,Yes,Mailed check,83.90,267.4
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
7262,9987-LUTYD,No,Female,0,No,No,13,Yes,No,DSL,...,No,No,Yes,No,No,One year,No,Mailed check,55.15,742.9
7263,9992-RRAMN,Yes,Male,0,Yes,No,22,Yes,Yes,Fiber optic,...,No,No,No,No,Yes,Month-to-month,Yes,Electronic check,85.10,1873.7
7264,9992-UJOEL,No,Male,0,No,No,2,Yes,No,DSL,...,Yes,No,No,No,No,Month-to-month,Yes,Mailed check,50.30,92.75
7265,9993-LHIEB,No,Male,0,Yes,Yes,67,Yes,No,DSL,...,No,Yes,Yes,No,Yes,Two year,No,Mailed check,67.85,4627.65


In [10]:
df_final.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7267 non-null   object 
 1   Churn             7267 non-null   object 
 2   gender            7267 non-null   object 
 3   SeniorCitizen     7267 non-null   int64  
 4   Partner           7267 non-null   object 
 5   Dependents        7267 non-null   object 
 6   tenure            7267 non-null   int64  
 7   PhoneService      7267 non-null   object 
 8   MultipleLines     7267 non-null   object 
 9   InternetService   7267 non-null   object 
 10  OnlineSecurity    7267 non-null   object 
 11  OnlineBackup      7267 non-null   object 
 12  DeviceProtection  7267 non-null   object 
 13  TechSupport       7267 non-null   object 
 14  StreamingTV       7267 non-null   object 
 15  StreamingMovies   7267 non-null   object 
 16  Contract          7267 non-null   object 


In [11]:
# Convertimos Charges.Total a numérico
# errors='coerce' reemplaza cualquier texto inválido por NaN (nulo)
df_final['Charges.Total'] = pd.to_numeric(df_final['Charges.Total'], errors='coerce')

In [12]:
# Verificamos si se crearon nulos durante la conversión
print(f"Nulos encontrados en Charges.Total: {df_final['Charges.Total'].isnull().sum()}")

Nulos encontrados en Charges.Total: 11


In [13]:
# Si hay nulos, podemos llenarlos con 0 (asumiendo que son clientes nuevos sin cargos totales aún)
df_final['Charges.Total'] = df_final['Charges.Total'].fillna(0)

In [14]:
# Verificamos nuevamente si apun hay nulos en Charges.Total
print(f"Nulos encontrados en Charges.Total: {df_final['Charges.Total'].isnull().sum()}")

Nulos encontrados en Charges.Total: 0


In [15]:
# Comparamos el promedio de cargos mensuales entre los que se quedan y los que se van
analisis_gastos = df_final.groupby('Churn')['Charges.Monthly'].mean()
print(analisis_gastos)

Churn
       63.412277
No     61.265124
Yes    74.441332
Name: Charges.Monthly, dtype: float64


In [18]:
df_final['Churn']

,Churn
0,No
1,No
2,Yes
3,Yes
4,Yes
...,...
7262,No
7263,Yes
7264,No
7265,No


In [19]:
# Filtramos el DataFrame donde la columna Churn sea exactamente un espacio en blanco
vacios_churn = df_final[df_final['Churn'] == " "]

In [20]:
# Mostramos cuántos hay y el listado de los primeros 10
print(f"Total de registros con Churn vacío: {len(vacios_churn)}")
vacios_churn[['customerID', 'tenure', 'Charges.Monthly', 'Churn']].head(10)

Total de registros con Churn vacío: 0


,customerID,tenure,Charges.Monthly,Churn


In [21]:
# Quitamos espacios invisibles a los lados de cada texto
df_final['Churn'] = df_final['Churn'].str.strip()

In [22]:
# Reemplazamos los strings que quedaron totalmente vacíos por NaN (nulo real)
import numpy as np
df_final['Churn'] = df_final['Churn'].replace('', np.nan)

In [23]:
# Ahora sí, eliminamos los nulos
df_final = df_final.dropna(subset=['Churn']).copy()

# Verificación final: Repetir el groupby
print("--- Grupos actuales en Churn ---")
print(df_final.groupby('Churn')['Charges.Monthly'].mean())

--- Grupos actuales en Churn ---
Churn
No     61.265124
Yes    74.441332
Name: Charges.Monthly, dtype: float64


In [24]:
print(df_final.isnull().sum())

customerID          0
Churn               0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
Charges.Monthly     0
Charges.Total       0
dtype: int64


In [25]:
df_final.shape

(7043, 21)

In [26]:
#Cuentas diarias
# Creamos la nueva columna dividiendo el cargo mensual por 30
df_final['Cuentas_Diarias'] = (df_final['Charges.Monthly'] / 30).round(2)

In [27]:
# Verificamos los primeros registros para ver la nueva columna
df_final[['customerID', 'Charges.Monthly', 'Cuentas_Diarias']].head()

,customerID,Charges.Monthly,Cuentas_Diarias
0,0002-ORFBO,65.6,2.19
1,0003-MKNFE,59.9,2.00
2,0004-TLHLJ,73.9,2.46
3,0011-IGKFF,98.0,3.27
4,0013-EXCHZ,83.9,2.80


In [28]:
# Comparamos el gasto diario promedio por estado de Churn
print("Gasto diario promedio:")
print(df_final.groupby('Churn')['Cuentas_Diarias'].mean())

Gasto diario promedio:
Churn
No     2.04208
Yes    2.48145
Name: Cuentas_Diarias, dtype: float64


El Diagnóstico de "Cuentas_Diarias"

Clientes que se quedan (No): Pagan en promedio $2.04 al día.

Clientes que se van (Yes): Pagaban en promedio $2.48 al día.

¿Qué significa esto para Telecom X? Que los clientes que se fugan están pagando, en promedio, un 21.5% más cada día que los que deciden quedarse. Esto refuerza la hipótesis de que el precio es un detonante crítico para la evasión.

In [29]:
# Seleccionamos las columnas numéricas clave
columnas_num = ['tenure', 'Charges.Monthly', 'Charges.Total', 'Cuentas_Diarias']

# Generamos el análisis descriptivo
analisis_stats = df_final[columnas_num].describe()

# Añadimos la mediana explícitamente (que es el percentil 50)
print("Análisis Descriptivo de Clientes:")
display(analisis_stats)

Análisis Descriptivo de Clientes:


,tenure,Charges.Monthly,Charges.Total,Cuentas_Diarias
count,7043.000000,7043.000000,7043.000000,7043.000000
mean,32.371149,64.761692,2279.734304,2.158675
std,24.559481,30.090047,2266.794470,1.003088
min,0.000000,18.250000,0.000000,0.610000
25%,9.000000,35.500000,398.550000,1.180000
50%,29.000000,70.350000,1394.550000,2.340000
75%,55.000000,89.850000,3786.600000,2.990000
max,72.000000,118.750000,8684.800000,3.960000


In [30]:
import plotly.express as px

# Preparamos los datos contando las categorías
churn_data = df_final['Churn'].value_counts().reset_index()
churn_data.columns = ['Estado', 'Total']

# Creamos el gráfico de dona
fig_pie = px.pie(churn_data,
                 values='Total',
                 names='Estado',
                 hole=0.5,
                 title='<b>Distribución Porcentual de Evasión (Churn)</b>',
                 color='Estado',
                 color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'}) # Verde para permanencia, Rojo para baja

fig_pie.update_traces(textinfo='percent+label', pull=[0, 0.1])
fig_pie.show()

In [31]:
# Gráfico de barras para ver cantidades exactas
fig_bar = px.bar(churn_data,
                 x='Estado',
                 y='Total',
                 text='Total',
                 color='Estado',
                 title='<b>Conteo Total de Clientes: Permanencia vs. Baja</b>',
                 labels={'Total': 'Cantidad de Clientes', 'Estado': '¿Hubo Evasión?'},
                 color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'},
                 template="plotly_white")

fig_bar.update_traces(textposition='outside')
fig_bar.show()

Gráfico de Proporción (Donut Chart)
Este es ideal para ver el "market share" interno de tus clientes. Nos dice qué porcentaje del total se ha ido.

In [32]:
import plotly.express as px

# 1. Contamos cuántos se quedaron (No) y cuántos se fueron (Yes)
churn_dist = df_final['Churn'].value_counts().reset_index()
churn_dist.columns = ['Estado', 'Total']

# 2. Creamos el gráfico de dona
fig_dona = px.pie(churn_dist,
                  values='Total',
                  names='Estado',
                  hole=0.5,
                  title='<b>Distribución Porcentual de Evasión (Churn)</b>',
                  color='Estado',
                  color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'}) # Verde=Fieles, Rojo=Fuga

fig_dona.update_traces(textinfo='percent+label')
fig_dona.show()

In [33]:
# Gráfico de barras interactivo
fig_barras = px.bar(churn_dist,
                    x='Estado',
                    y='Total',
                    text='Total',
                    color='Estado',
                    title='<b>Cantidad Total de Clientes por Estado</b>',
                    labels={'Total': 'Número de Clientes', 'Estado': '¿Se dio de baja?'},
                    color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'},
                    template="plotly_white")

fig_barras.update_traces(textposition='outside')
fig_barras.show()

In [34]:
import plotly.express as px

# 1. Creamos una tabla de frecuencias para el gráfico
df_contrato = df_final.groupby(['Contract', 'Churn']).size().reset_index(name='Cantidad')

# 2. Generamos el gráfico de barras apiladas (Stacked Bar Chart)
fig_contrato = px.bar(df_contrato,
                     x='Contract',
                     y='Cantidad',
                     color='Churn',
                     title='<b>Análisis de Evasión por Tipo de Contrato</b>',
                     labels={'Cantidad': 'Número de Clientes', 'Contract': 'Tipo de Contrato'},
                     barmode='stack',
                     color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'},
                     template="plotly_white")

fig_contrato.show()

In [35]:
#Churn por tipo de contrato
#Comparativa visual de barras apiladas

import plotly.express as px

# 1. Creamos una tabla de frecuencias para el gráfico
df_contrato = df_final.groupby(['Contract', 'Churn']).size().reset_index(name='Cantidad')

# 2. Generamos el gráfico de barras apiladas (Stacked Bar Chart)
fig_contrato = px.bar(df_contrato,
                     x='Contract',
                     y='Cantidad',
                     color='Churn',
                     title='<b>Análisis de Evasión por Tipo de Contrato</b>',
                     labels={'Cantidad': 'Número de Clientes', 'Contract': 'Tipo de Contrato'},
                     barmode='stack',
                     color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'},
                     template="plotly_white")

fig_contrato.show()

El "Dato más relevante":la barra de "Month-to-month" (mes a mes) tiene la mayor proporción de color rojo (Yes). Esto indica que los clientes sin compromiso a largo plazo son los más propensos a evadir.

La Barrera de Salida: En los contratos de "One year" o "Two year", el color rojo debería ser mínimo. Esto demuestra que los contratos largos actúan como un "seguro" de retención para la empresa.

Volumen de Riesgo: en el tamaño total de la barra "Month-to-month" es la más alta de todas, significa que la mayoría de los clientes de Telecom X están en la categoría de alto riesgo.





In [36]:
#Cruce de Variables Categóricas
#Para este análisis,vamos a usar Gráficos de Barras Agrupados. Comparar el "Yes" y el "No" en tres ejes críticos: Género, Método de Pago y Tipo de Contrato.

import plotly.express as px

# Definimos las variables categóricas que queremos explorar
variables_cat = ['gender', 'Contract', 'PaymentMethod', 'InternetService']

for var in variables_cat:
    # Creamos un DataFrame agrupado para cada variable
    df_grouped = df_final.groupby([var, 'Churn']).size().reset_index(name='Cantidad')

    # Generamos el gráfico
    fig = px.bar(df_grouped,
                 x=var,
                 y='Cantidad',
                 color='Churn',
                 barmode='group',
                 title=f'<b>Distribución de Evasión por {var}</b>',
                 color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'},
                 template="plotly_white")
    fig.show()

In [37]:
# Para Método de Pago en porcentajes
print("Tasa de Evasión por Método de Pago (%):")
print(pd.crosstab(df_final['PaymentMethod'], df_final['Churn'], normalize='index') * 100)

Tasa de Evasión por Método de Pago (%):
Churn                             No        Yes
PaymentMethod                                  
Bank transfer (automatic)  83.290155  16.709845
Credit card (automatic)    84.756899  15.243101
Electronic check           54.714588  45.285412
Mailed check               80.893300  19.106700


In [38]:
#análisis por antigüedad
import plotly.express as px

# Creamos un histograma interactivo para ver la distribución de meses
fig_tenure = px.histogram(df_final,
                          x="tenure",
                          color="Churn",
                          marginal="box", # Añade un diagrama de caja arriba para ver la mediana
                          title="<b>Distribución de Antigüedad: ¿Cuándo se van los clientes?</b>",
                          labels={'tenure': 'Meses de Antigüedad', 'count': 'Número de Clientes'},
                          color_discrete_map={'No': '#2ECC71', 'Yes': '#E74C3C'},
                          template="plotly_white",
                          barmode='overlay')

fig_tenure.show()

"Existe una correlación crítica entre el método de pago y la evasión. Mientras que los pagos automáticos mantienen tasas de fuga bajas (aprox. 15-16%), el uso de 'Electronic check' dispara la evasión al 45.28%. Esto sugiere que automatizar el cobro es una de las estrategias de retención más efectivas para Telecom X."

El Problema: Fibra Óptica (Fiber optic)
Es el hallazgo más relevante:

Proporción Crítica: En la categoría de Fiber optic, la barra roja (Yes) es casi tan alta como la verde (No).

Contraste con DSL: Mientras que en DSL la evasión es mínima comparada con los clientes que se quedan, en Fibra Óptica la tasa de fuga es masiva.

Relación con Costos: Recordemos que el gasto diario de los que se van es mayor ($2.48) que el de los que se quedan ($2.04). Esto sugiere que la Fibra Óptica es más cara y, posiblemente, los clientes no sienten que la calidad justifique el precio, o están experimentando fallas técnicas.

Segmento Estable: DSL y Sin Internet
DSL: Tiene una base de clientes muy leal. La barra verde es dominante, lo que indica que es un servicio maduro y estable.

No (Sin Internet): Los clientes que solo tienen telefonía casi no se van. Esto confirma que el problema de evasión está directamente ligado a los servicios de datos de alta velocidad.

el género no es un factor determinante para la evasión.

Volumen de Evasión: La gran mayoría de los clientes que se van (Churn=Yes) pertenecen al contrato Month-to-month, con un total de 1,655 clientes en este grupo.

Comparativa de Estabilidad: En los contratos de One year y Two year, la evasión es casi inexistente en comparación con los clientes que permanecen.

Insight: El contrato mensual no solo ofrece libertad al cliente, sino que parece ser el refugio de los usuarios más insatisfechos o sensibles al precio.

El Factor Económico (Cuentas Diarias)
Al conectar esto con el cálculo de gastos, vemos el "porqué" del abandono:

Costo de Evasión: El gasto diario promedio de quienes se van es de 2.48.

Costo de Retención: Los clientes que se quedan pagan en promedio 2.04.

Interpretación: Los clientes con contratos mensuales suelen tener tarifas más altas (sin los descuentos de los contratos anuales). Ese costo extra de 0.44 diarios (un 21.5% más) es el motor que los impulsa a cancelar el servicio.



"El perfil de riesgo en Telecom X es un cliente de Fibra Óptica con contrato mes a mes que paga mediante cheque electrónico. Estos usuarios enfrentan los costos diarios más altos (2.48) y suelen abandonar la compañía en sus primeros meses de servicio. La estrategia debe enfocarse en automatizar sus pagos y ofrecerles descuentos por migrar a contratos anuales."


# ******Informe Final: Análisis de Evasión (Churn) - Telecom X *********

## 1. Introducción

El objetivo de este análisis es identificar los factores críticos que influyen en ***la evasión de clientes (Churn) de Telecom X***. El Churn representa el porcentaje de suscriptores que cancelan su servicio, lo cual impacta directamente en la rentabilidad de la empresa, ya que el costo de adquisición de un nuevo cliente es significativamente mayor que el de retención.

## 2. Limpieza y Tratamiento de Datos

Para garantizar la calidad del análisis, se realizaron los siguientes pasos:

* **Importación:** Se cargaron los datos desde el archivo JSON proporcionado, el cual contenía información demográfica, de servicios y financiera de 7,043 clientes.
* **Tratamiento de Nulos:** Se identificaron y eliminaron filas con valores vacíos en la columna "Charges.Total", asegurando un dataset final limpio de **7,043 registros**.
* **Ingeniería de Datos:** Se creó la columna **Cuentas_Diarias**, dividiendo el cargo mensual por 30 para obtener una visión más detallada del valor diario de cada cliente.

## 3. Análisis Exploratorio de Datos (EDA)

Durante la exploración, se identificaron patrones clave mediante visualizaciones:

* **Distribución Financiera:** Los clientes que abandonan la compañía pagan, en promedio, **$2.48 diarios**, frente a los **$2.04** de los clientes que permanecen.
* **Impacto del Contrato:** El contrato **"Month-to-month"** presenta el mayor volumen de fugas con **1,655 bajas**, mientras que los contratos anuales muestran una estabilidad casi total.
* **Riesgo por Servicio:** El servicio de **Fibra Óptica** tiene una tasa de evasión alarmante, con casi tantos clientes retirándose como permaneciendo, a diferencia del servicio DSL que es muy estable.
* **Antigüedad:** El riesgo de fuga es crítico en los **primeros 6 meses**. El gráfico de Boxplot muestra que la mediana de antigüedad para los que se van es de apenas **10 meses**.

## 4. Conclusiones e Insights

* **Sensibilidad al Precio:** Los clientes que evaden enfrentan cargos diarios un **21.5% superiores** a los de los clientes retenidos.
* **Falta de Compromiso:** El método de pago **"Electronic check"** tiene la tasa de evasión más alta con un **45.28%**.
* **Neutralidad Demográfica:** El género no es un factor determinante, ya que la distribución de fuga es casi idéntica entre hombres y mujeres.

## 5. Recomendaciones Estratégicas

1. **Fidelización Temprana:** Implementar campañas de "Acompañamiento" durante los primeros 12 meses, que es el periodo de mayor riesgo según el análisis de *tenure*.
2. **Incentivos Contractuales:** Ofrecer descuentos agresivos a los clientes de "Fibra Óptica" que actualmente tienen contrato mes a mes para que migren a contratos de 1 o 2 años.
3. **Automatización de Pagos:** Promover el uso de tarjeta de crédito o transferencia automática, dado que estos métodos tienen tasas de evasión 3 veces menores que el cheque electrónico.
4. **Revisión de Fibra Óptica:** Investigar la calidad técnica o competitividad de precios de la Fibra Óptica, ya que es el servicio con mayor pérdida de usuarios.
